awx/main/migrations/0183_pre_django_upgrade.py

```diff
@@ -0,0 +1,30 @@
+# Generated by Django 3.2.16 on 2023-04-21 14:15
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        migrations.swappable_dependency(settings.OAUTH2_PROVIDER_ID_TOKEN_MODEL),
+        ('main', '0182_constructed_inventory'),
+        ('oauth2_provider', '0005_auto_20211222_2352'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='oauth2accesstoken',
+            name='id_token',
+            field=models.OneToOneField(
+                blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='access_token', to=settings.OAUTH2_PROVIDER_ID_TOKEN_MODEL
+            ),
+        ),
+        migrations.AddField(
+            model_name='oauth2application',
+            name='algorithm',
+            field=models.CharField(
+                blank=True, choices=[('', 'No OIDC support'), ('RS256', 'RSA with SHA-2 256'), ('HS256', 'HMAC with SHA-2 256')], default='', max_length=5
+            ),
+        ),
+    ]
```

Comment:
For these 2 migration files, it might be helpful to share the `awx-manage sqlmigrate` output. This will help quickly clear up which steps are and are not making a schema change. I mostly expect to see non-schema changes as a result of a Django upgrade, as its own internal accounting changes.

Reply:
0183 is an oauth-toolkit migration and has legitimate output:
```
BEGIN;
--
-- Add field id_token to oauth2accesstoken
--
ALTER TABLE "main_oauth2accesstoken" ADD COLUMN "id_token_id" bigint NULL UNIQUE CONSTRAINT "main_oauth2accesstok_id_token_id_8a01d0b0_fk_oauth2_pr" REFERENCES "oauth2_provider_idtoken"("id") DEFERRABLE INITIALLY DEFERRED; SET CONSTRAINTS "main_oauth2accesstok_id_token_id_8a01d0b0_fk_oauth2_pr" IMMEDIATE;
--
-- Add field algorithm to oauth2application
--
ALTER TABLE "main_oauth2application" ADD COLUMN "algorithm" varchar(5) DEFAULT '' NOT NULL;
ALTER TABLE "main_oauth2application" ALTER COLUMN "algorithm" DROP DEFAULT;
COMMIT;
```

0184 is from the reindexing and, like you mentioned, all no-ops:
```
BEGIN;
--
-- Rename unnamed index for ('ad_hoc_command', 'job_created', 'uuid') on adhoccommandevent to main_adhocc_ad_hoc__1e4d24_idx
--
-- (no-op)
--
-- Rename unnamed index for ('ad_hoc_command', 'job_created', 'event') on adhoccommandevent to main_adhocc_ad_hoc__e72142_idx
--
-- (no-op)
--
-- Rename unnamed index for ('ad_hoc_command', 'job_created', 'counter') on adhoccommandevent to main_adhocc_ad_hoc__a57777_idx
--
-- (no-op)
--
-- Rename unnamed index for ('inventory_update', 'job_created', 'uuid') on inventoryupdateevent to main_invent_invento_f72b21_idx
--
-- (no-op)
--
-- Rename unnamed index for ('inventory_update', 'job_created', 'counter') on inventoryupdateevent to main_invent_invento_364dcb_idx
--
-- (no-op)
--
-- Rename unnamed index for ('job', 'job_created', 'parent_uuid') on jobevent to main_jobeve_job_id_40a56d_idx
--
-- (no-op)
--
-- Rename unnamed index for ('job', 'job_created', 'uuid') on jobevent to main_jobeve_job_id_3c4a4a_idx
--
-- (no-op)
--
-- Rename unnamed index for ('job', 'job_created', 'counter') on jobevent to main_jobeve_job_id_51c382_idx
--
-- (no-op)
--
-- Rename unnamed index for ('job', 'job_created', 'event') on jobevent to main_jobeve_job_id_0ddc6b_idx
--
-- (no-op)
--
-- Rename unnamed index for ('project_update', 'job_created', 'uuid') on projectupdateevent to main_projec_project_449bbd_idx
--
-- (no-op)
--
-- Rename unnamed index for ('project_update', 'job_created', 'counter') on projectupdateevent to main_projec_project_69559a_idx
--
-- (no-op)
--
-- Rename unnamed index for ('project_update', 'job_created', 'event') on projectupdateevent to main_projec_project_c44b7c_idx
--
-- (no-op)
--
-- Rename unnamed index for ('content_type', 'object_id') on role to main_rbac_r_content_979bdd_idx
--
-- (no-op)
--
-- Rename unnamed index for ('ancestor', 'content_type_id', 'role_field') on roleancestorentry to main_rbac_r_ancesto_b44606_idx
--
-- (no-op)
--
-- Rename unnamed index for ('ancestor', 'content_type_id', 'object_id') on roleancestorentry to main_rbac_r_ancesto_22b9f0_idx
--
-- (no-op)
--
-- Rename unnamed index for ('ancestor', 'descendent') on roleancestorentry to main_rbac_r_ancesto_c87b87_idx
--
-- (no-op)
--
-- Rename unnamed index for ('system_job', 'job_created', 'uuid') on systemjobevent to main_system_system__e39825_idx
--
-- (no-op)
--
-- Rename unnamed index for ('system_job', 'job_created', 'counter') on systemjobevent to main_system_system__73537a_idx
--
-- (no-op)
--
-- Alter field unified_job on activitystream
--
-- (no-op)
--
-- Alter field unified_job_template on activitystream
--
-- (no-op)
--
-- Alter field created_by on credential
--
-- (no-op)
--
-- Alter field modified_by on credential
--
-- (no-op)
--
-- Alter field created_by on credentialinputsource
--
-- (no-op)
--
-- Alter field modified_by on credentialinputsource
--
-- (no-op)
--
-- Alter field created_by on credentialtype
--
-- (no-op)
--
-- Alter field modified_by on credentialtype
--
-- (no-op)
--
-- Alter field created_by on custominventoryscript
--
-- (no-op)
--
-- Alter field modified_by on custominventoryscript
--
-- (no-op)
--
-- Alter field created_by on executionenvironment
--
-- (no-op)
--
-- Alter field credential on executionenvironment
--
-- (no-op)
--
-- Alter field modified_by on executionenvironment
--
-- (no-op)
--
-- Alter field organization on executionenvironment
--
-- (no-op)
--
-- Alter field created_by on group
--
-- (no-op)
--
-- Alter field modified_by on group
--
-- (no-op)
--
-- Alter field created_by on host
--
-- (no-op)
--
-- Alter field modified_by on host
--
-- (no-op)
--
-- Alter field smart_inventories on host
--
-- (no-op)
--
-- Alter field credential on instancegroup
--
-- (no-op)
--
-- Alter field created_by on inventory
--
-- (no-op)
--
-- Alter field modified_by on inventory
--
-- (no-op)
--
-- Alter field inventory on job
--
-- (no-op)
--
-- Alter field project on job
--
-- (no-op)
--
-- Alter field webhook_credential on job
--
-- (no-op)
--
-- Alter field credentials on joblaunchconfig
--
-- (no-op)
--
-- Alter field execution_environment on joblaunchconfig
--
-- (no-op)
--
-- Alter field instance_groups on joblaunchconfig
--
-- (no-op)
--
-- Alter field inventory on joblaunchconfig
--
-- (no-op)
--
-- Alter field labels on joblaunchconfig
--
-- (no-op)
--
-- Alter field inventory on jobtemplate
--
-- (no-op)
--
-- Alter field project on jobtemplate
--
-- (no-op)
--
-- Alter field webhook_credential on jobtemplate
--
-- (no-op)
--
-- Alter field created_by on label
--
-- (no-op)
--
-- Alter field modified_by on label
--
-- (no-op)
--
-- Alter field created_by on notificationtemplate
--
-- (no-op)
--
-- Alter field modified_by on notificationtemplate
--
-- (no-op)
--
-- Alter field user on oauth2accesstoken
--
-- (no-op)
--
-- Alter field user on oauth2application
--
-- (no-op)
--
-- Alter field created_by on organization
--
-- (no-op)
--
-- Alter field galaxy_credentials on organization
--
-- (no-op)
--
-- Alter field modified_by on organization
--
-- (no-op)
--
-- Alter field notification_templates_approvals on organization
--
-- (no-op)
--
-- Alter field notification_templates_error on organization
--
-- (no-op)
--
-- Alter field notification_templates_started on organization
--
-- (no-op)
--
-- Alter field notification_templates_success on organization
--
-- (no-op)
--
-- Alter field credential on project
--
-- (no-op)
--
-- Alter field signature_validation_credential on project
--
-- (no-op)
--
-- Alter field credential on projectupdate
--
-- (no-op)
--
-- Alter field created_by on schedule
--
-- (no-op)
--
-- Alter field credentials on schedule
--
-- (no-op)
--
-- Alter field execution_environment on schedule
--
-- (no-op)
--
-- Alter field inventory on schedule
--
-- (no-op)
--
-- Alter field labels on schedule
--
-- (no-op)
--
-- Alter field modified_by on schedule
--
-- (no-op)
--
-- Alter field created_by on team
--
-- (no-op)
--
-- Alter field modified_by on team
--
-- (no-op)
--
-- Alter field created_by on unifiedjob
--
-- (no-op)
--
-- Alter field credentials on unifiedjob
--
-- (no-op)
--
-- Alter field dependent_jobs on unifiedjob
--
-- (no-op)
--
-- Alter field execution_environment on unifiedjob
--
-- (no-op)
--
-- Alter field labels on unifiedjob
--
-- (no-op)
--
-- Alter field modified_by on unifiedjob
--
-- (no-op)
--
-- Alter field notifications on unifiedjob
--
-- (no-op)
--
-- Alter field organization on unifiedjob
--
-- (no-op)
--
-- Alter field polymorphic_ctype on unifiedjob
--
-- (no-op)
--
-- Alter field unified_job_template on unifiedjob
--
-- (no-op)
--
-- Alter field created_by on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field credentials on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field current_job on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field execution_environment on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field labels on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field last_job on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field modified_by on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field next_schedule on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field notification_templates_error on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field notification_templates_started on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field notification_templates_success on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field organization on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field polymorphic_ctype on unifiedjobtemplate
--
-- (no-op)
--
-- Alter field approved_or_denied_by on workflowapproval
--
-- (no-op)
--
-- Alter field inventory on workflowjob
--
-- (no-op)
--
-- Alter field webhook_credential on workflowjob
--
-- (no-op)
--
-- Alter field always_nodes on workflowjobnode
--
-- (no-op)
--
-- Alter field credentials on workflowjobnode
--
-- (no-op)
--
-- Alter field execution_environment on workflowjobnode
--
-- (no-op)
--
-- Alter field failure_nodes on workflowjobnode
--
-- (no-op)
--
-- Alter field inventory on workflowjobnode
--
-- (no-op)
--
-- Alter field labels on workflowjobnode
--
-- (no-op)
--
-- Alter field success_nodes on workflowjobnode
--
-- (no-op)
--
-- Alter field unified_job_template on workflowjobnode
--
-- (no-op)
--
-- Alter field inventory on workflowjobtemplate
--
-- (no-op)
--
-- Alter field notification_templates_approvals on workflowjobtemplate
--
-- (no-op)
--
-- Alter field webhook_credential on workflowjobtemplate
--
-- (no-op)
--
-- Alter field always_nodes on workflowjobtemplatenode
--
-- (no-op)
--
-- Alter field credentials on workflowjobtemplatenode
--
-- (no-op)
--
-- Alter field execution_environment on workflowjobtemplatenode
--
-- (no-op)
--
-- Alter field failure_nodes on workflowjobtemplatenode
--
-- (no-op)
--
-- Alter field inventory on workflowjobtemplatenode
--
-- (no-op)
--
-- Alter field labels on workflowjobtemplatenode
--
-- (no-op)
--
-- Alter field success_nodes on workflowjobtemplatenode
--
-- (no-op)
--
-- Alter field unified_job_template on workflowjobtemplatenode
--
-- (no-op)
COMMIT;
```